SIPS/sip-12.md

```diff
@@ -53,22 +53,20 @@ The proposal for Dynamic gas pricing is as follows:
 Implementing a maximum gas price on exchange transactions and setting it just below the gas price of oracle update transactions will prevent front running and minimize disruption to legitimate exchanges (for normal users who don't need their exchange to be mined immediately and can wait a few blocks).
 
 The example below illustrates how this mechanism will function:
-1. An oracle updates the current "Fast" and "Fastest" gas price to 20 gwei and 25 gwei respectively
-2. The Synthetix contract reads this and updates the max gas price for exchanges to a weighted AVG (configurable via SCCP) between Fast and Fastest
-3. The synthetix.exchange dApp reads the current max gas price of 23.5 gwei and sets this as the max gas price
-4. The exchange rates oracle updates its gas price to 125% (configurable via SCCP) of Fastest, 31.25 gwei
-5. A frontrunning bot detects a spot market deviation of >.3% (assuming a fee of 30bps)
-6. The Oracle detects the same price deviation
-7. Both txs are broadcast simultaneously, the exchange at 23.5 and the rate update at 31.25
-8. The rate update confirms first ensuring that the frontrunning bot always gets he current live rate from the spot market
+
+1. An oracle updates the current `fast` gas price to `20` gwei on-chain.
+2. A frontrunning bot detects a spot market deviation of `>.3%` (assuming a fee of 30bps). It issues an exchange at the highest GWEI allowed by the `Synthetix` contract, which is `20` from above.
+3. The SNX Oracle reads both `fast` on-chain (`20`) and `fast` off-chain via gas station APIs (let's say it has since dropped to `15`). It updates its gas price to `125%` (configurable via SCCP) of `Math.max(fast on-chain, fast off-chain)`. Which is `25` gwei
+4. Both txs are broadcast simultaneously, the exchange bot at `20` and the SNX Oracle rate update at `25` gwei
+5. The rate update confirms first ensuring that the frontrunning bot always gets the current live rate from the spot market
```

Comment:
I think Point 5 can be explained more clearly as a speedbump for bots so it would be harder to frontrun the exchangeRate prices.

Reply:
Want to suggest some new wording?

Reply:
What wording do you suggest? The speedbump analogy makes sense when you highlight the fact that the oracle and the bot are on different timing schedules (and unlikely to transact at the exact same moment), though that means writing up a whole new example.

Reply:
Fair enough, I think the phrase `the rate update confirms first` covers that the oracle rates will be there first before traders / frontrunning bots can take an advantage position ahead of the oracle.